docs/proposals/dynamic-app-parameters.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Dynamic Application Parameters
+authors:
+  - "@jannfis" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2023-01-17
+last-updated: 2023-01-17
+---
+
+# Dynamic application parameters
+
+## Open questions
+
+* Should the value of a dynamic parameter be visible to a user who has `get` permissions of an Application (see also Security Implications below)?
+
+## Summary
+
+This document proposes the introduction of a new type of application parameters which are resolved dynamically at manifest rendering time. Its values will be retrieved from the application's destination cluster. Details of this are described in this proposal.
+
+## Motivation
```

Comment:
Might another potential motivation be to provide enhanced functionality for Kustomize similar to that offered by Flux via its `PostBuild SubstitutionFrom` feaure described [here](https://fluxcd.io/flux/components/kustomize/api/#kustomize.toolkit.fluxcd.io/v1beta2.PostBuild)?